kornia/augmentation/random.py

```diff
@@ -148,18 +148,19 @@ def _get_perspective_params(batch_size: int, width: int, height: int, distortion
     ]]).expand(batch_size, -1, -1)
 
     # generate random offset not larger than half of the image
-    fx: float = distortion_scale * width / 2
-    fy: float = distortion_scale * height / 2
-
-    factor = torch.tensor([[
-        [fy, fx],
-        [fy, width - fx - 1],
-        [height - fy - 1, fx],
-        [height - fy - 1, width - fx - 1],
-    ]]).expand(batch_size, -1, -1)
-
-    rand_val: torch.Tensor = _adapted_uniform((batch_size, 4, 2), 1 - distortion_scale, 1, same_on_batch)
-    end_points = factor * rand_val
+    fx: float = distortion_scale * height / 2
+    fy: float = distortion_scale * width / 2
+
+    factor = torch.tensor([fx, fy]).view(-1, 1, 2)
+
+    rand_val: torch.Tensor = _adapted_uniform((batch_size, 4, 2), 0, 1, same_on_batch)
+    mat = torch.tensor([[
```

Comment:
maybe `pts_norm` it's a better name for this variable

Reply:
I think about this and usually  when using normalized coordinates top-left [-1, -1] and bottom-right [1, 1]. If we do it in that way we should follow same convention as we do in other places (and pytorch does with grid_sample).
```
fx: float = distortion_scale * width / 2
fy: float = distortion_scale * height / 2

factor = torch.tensor([fx, fy]).view(-1, 1, 2)

# order for each point must be [x, y]
mat = torch.tensor([[
  [-1, -1],  # top-left
  [1, -1],   # top-right
  [-1, -1],  # bottom-right
  [-1, 1]    # bottom-left
]])

Reply:
In this case, it will not align with TorchVision.
https://pytorch.org/docs/stable/_modules/torchvision/transforms/transforms.html#RandomPerspective

Reply:
ok, then you have to swap x/y order when you call apply_perspective, otherwise the math in [`get_perspective_transform`](https://github.com/kornia/kornia/blob/master/kornia/geometry/transform/imgwarp.py#L260-L282) won't work